src/SFML/Audio/SoundFileWriterOgg.cpp

```diff
@@ -32,8 +32,15 @@
 #include <algorithm>
 #include <cassert>
 #include <cctype>
-#include <cstdlib>
 #include <ostream>
+#include <random>
+
+
+namespace
+{
+std::random_device rd;
+std::mt19937       rng(rd());
+} // namespace
```

Comment:
Is this save to initialize at global startup, or could these types depend on global objects initialized somewhere else, potentially at a later stage?

Also a more general question, do we want to advocate having RNGs as global variables? Examples are also used to highlight best practices from our point of view.

Reply:
> Is this save to initialize at global startup, or could these types depend on global objects initialized somewhere else, potentially at a later stage?

We use this pattern elsewhere in SFML and I've used it in many other projects with no issue so I believe it's perfectly safe.

> Also a more general question, do we want to advocate having RNGs as global variables? Examples are also used to highlight best practices from our point of view.

The alternative is to use a `static` local variable which I'm fine doing in this case if you prefer.

Reply:
```suggestion
std::mt19937& rng() 
{
    static std::mt19937 result{std::random_device{}()};    
    return result;
}
```

Or maybe `thread_local`? Do we care about thread-safety here?